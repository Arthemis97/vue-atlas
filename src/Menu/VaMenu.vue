<template>
  <nav
    :class="`${classPrefix}-menu`"
    role="menu"
    outline="none">
    <slot/>
  </nav>
</template>

<script>
export default {
  name: 'VaMenu',
  props: {
    classPrefix: {
      type: String,
      default: 'va'
    }
  }
}
</script>
