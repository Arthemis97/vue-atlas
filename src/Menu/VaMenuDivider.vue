<template>
  <div :class="`${classPrefix}-menu__divider`"></div>
</template>

<script>
export default {
  name: 'VaMenuDivider',
  props: {
    classPrefix: {
      type: String,
      default: 'va'
    }
  }
}
</script>
