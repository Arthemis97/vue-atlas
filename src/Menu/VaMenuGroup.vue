<template>
  <div :class="`${classPrefix}-menu__group`">
    <slot />
  </div>
</template>

<script>
export default {
  name: 'VaMenuGroup',
  props: {
    classPrefix: {
      type: String,
      default: 'va'
    }
  }
}
</script>
