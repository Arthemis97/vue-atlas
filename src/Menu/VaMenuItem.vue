<template>
  <div
    :class="`${classPrefix}-menu__item`"
    role="menuitem"
    @click="onSelect">
    <span>
      <slot />
    </span>
  </div>
</template>

<script>
export default {
  name: 'VaMenuItem',
  props: {
    classPrefix: {
      type: String,
      default: 'va'
    }
  },
  methods: {
    onSelect (e) {
      this.$emit('select', e)
    }
  }
}
</script>
