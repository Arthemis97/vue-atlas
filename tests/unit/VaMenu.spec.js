import { shallowMount } from '@vue/test-utils'
import VaMenu from '@/Menu/VaMenu.vue'
import VaMenuGroup from '@/Menu/VaMenuGroup.vue'
import VaMenuItem from '@/Menu/VaMenuItem.vue'
import VaMenuDivider from '@/Menu/VaMenuDivider.vue'

describe('VaMenu.vue', () => {
  it('renders correctly with children', () => {
    const wrapper = shallowMount({
      render () {
        return (
          <VaMenu>
            <VaMenuGroup>
              <VaMenuItem>Item</VaMenuItem>
              <VaMenuItem>Item</VaMenuItem>
            </VaMenuGroup>
            <VaMenuDivider></VaMenuDivider>
            <VaMenuGroup>
              <VaMenuItem>Item</VaMenuItem>
            </VaMenuGroup>
          </VaMenu>
        )
      }
    })
    expect(wrapper).toMatchSnapshot()
  })
})