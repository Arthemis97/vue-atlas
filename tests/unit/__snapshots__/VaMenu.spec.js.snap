// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VaMenu.vue renders correctly with children 1`] = `
<nav role="menu" outline="none" class="va-menu">
  <div class="va-menu__group">
    <div role="menuitem" class="va-menu__item"><span>Item</span></div>
    <div role="menuitem" class="va-menu__item"><span>Item</span></div>
  </div>
  <div class="va-menu__divider"></div>
  <div class="va-menu__group">
    <div role="menuitem" class="va-menu__item"><span>Item</span></div>
  </div>
</nav>
`;
